src/Parser/Rogue/Subtlety/Modules/BaseRotation/CastsInShadowDance.js

```diff
@@ -0,0 +1,87 @@
+    damageTracker: DamageTracker,
+    danceDamageTracker: DanceDamageTracker,
+    combatants: Combatants,
+  };
+
+  suggestions(when) { 
+    
+    const filtered = this.danceDamageTracker.getAbility(SPELLS.BACKSTAB.id);
+    when(filtered).isGreaterThan(0)
+    .addSuggestion((suggest, actual, recommended) => {
+      return suggest(<Wrapper>Use <SpellLink id={SPELLS.SHADOWSTRIKE.id} /> instead of <SpellLink id={SPELLS.BACKSTAB.id} /> during <SpellLink id={SPELLS.SHADOW_DANCE.id} />. </Wrapper>)
+        .icon(SPELLS.BACKSTAB.icon)
+        .actual(`You cast Backstab ${filtered} times during Shadow Dance.`)
+        .recommended(`0 is recommend.`)
+        .regular(0).major(1);
+    });    
+    
+    this.suggestBackstab(when);
+    this.suggestGloomblade(when);
+    this.suggestAvgCasts(when);
+  }
+  
+  statistic() {
+    const shadowDanceUptime = this.combatants.selected.getBuffUptime(SPELLS.SHADOW_DANCE_BUFF.id) / this.owner.fightDuration;
+    return (
+      <StatisticBox
+        icon={<SpellIcon id={SPELLS.SHADOW_DANCE_BUFF.id} />}
+        value={`${formatPercentage(shadowDanceUptime)} %`}
+        label="Shadow Dance uptime"
+      />
+    );
+  }
+  statisticOrder = STATISTIC_ORDER.CORE(2);
+  
+  
+  suggestBackstab(when) {
+  }
+  suggestGloomblade(when) {
+    const filtered = this.danceDamageTracker.getAbility(SPELLS.GLOOMBLADE_TALENT.id);
+    when(filtered).isGreaterThan(0)
+    .addSuggestion((suggest, actual, recommended) => {
+    return suggest(<Wrapper>Use <SpellLink id={SPELLS.SHADOWSTRIKE.id} /> instead of <SpellLink id={SPELLS.GLOOMBLADE_TALENT.id} /> during <SpellLink id={SPELLS.SHADOW_DANCE.id} />. </Wrapper>)
+        .icon(SPELLS.GLOOMBLADE_TALENT.icon)
+        .actual(`You cast Gloomblade ${filtered} times during Shadow Dance.`)
+        .recommended(`0 is recommend.`)
+        .regular(0).major(1);
+    });  
+  }
+  suggestAvgCasts(when){
+    const danceCount = this.damageTracker.getAbility(SPELLS.SHADOW_DANCE.id);
+    const castsInDance = this.danceDamageTracker.getAbility(SPELLS.BACKSTAB.id).casts 
+    + this.danceDamageTracker.getAbility(SPELLS.GLOOMBLADE_TALENT.id).casts 
+    + this.danceDamageTracker.getAbility(SPELLS.SHURIKEN_STORM.id).casts 
+    + this.danceDamageTracker.getAbility(SPELLS.SHADOWSTRIKE.id).casts 
+    + this.danceDamageTracker.getAbility(SPELLS.NIGHTBLADE.id).casts 
+    + this.danceDamageTracker.getAbility(SPELLS.EVISCERATE.id).casts;
+
+    const missedCastsInDanceShare = castsInDance/(danceCount * 4);
+    when(missedCastsInDanceShare).isLessThan(1)
+    .addSuggestion((suggest, actual, recommended) => {
+      return suggest(<Wrapper>Try to cast 4 spells during each <SpellLink id={SPELLS.SHADOW_DANCE_BUFF.id} /> </Wrapper>)
+        .icon(SPELLS.SHADOW_DANCE_BUFF.icon)
+        .actual(`You cast less ${castsInDance} spells during Shadow Dance out of ${danceCount * 4} possible.`)
+        .recommended(`${danceCount * 4} recommended`)
+        .regular(0.90).major(0.80);
```

Comment:
The idea here is that you must cast exactly 4 times (max possible), but I don't want to have it major if its just one dance with 3. So I am checking the average casts to find out how bad it is but suggesting exactly 4.

Reply:
It would read better to say 4 casts per shadow dance is recommended, or something similiar. Giving me a multiplied number as recommended, doesn't really help me on the fly - whereas "4 pr cast" is quite tangible. 
The wording in "actual" is still bad and needs to be updated regardless.

Reply:
the wording is still off. Now you're suggestion 4 times shadowDance casts per shadow Dance.